Debug text prop not working in some cases

OyvindLGjesdal opened the pull request:
GitHub issue resolved #2094

Pull request Description:

* Add field grouping (parens) to composeQField method to fix #2094 

I see a similar fix with parens is also used in other paths in the code for example in line 765 and 771 of the same file.

```
   qString = "(" + qString + ") AND " + (!lang.equals("none") ? langField+":"+lang : "-"+langField+":*");
            log.trace("query$ ADDING LANG qString: {} ", qString) ;
        }
```

```
  if (graphURI != null) {
            String escaped = QueryParserBase.escape(graphURI) ;
            qString = "(" + qString + ") AND " + getDocDef().getGraphField() + ":" + escaped ;
        }
```

*  Add a unique set to remove duplicate query clauses on propLists which is mentioned by @rvesse 

> It also looks like we generate duplicate query clauses when multiple properties map to the same Lucene field which might be unnecessary?

----

 - [x] Tests are included.
 - [x] (Not applicable) Documentation change and updates are provided for the [Apache Jena website](https://github.com/apache/jena-site/)
 - [x] Commits have been squashed to remove intermediate development commit messages.
 - [x] Key commit messages start with the issue number (GH-xxxx, or if in JIRA, JENA-xxxx)

By submitting this pull request, I acknowledge that I am making a contribution to the Apache Software Foundation under the terms and conditions of the [Contributor's Agreement](https://www.apache.org/licenses/contributor-agreements.html).

----

See the [Apache Jena "Contributing" guide](https://github.com/apache/jena/blob/main/CONTRIBUTING.md).

rvesse commented on jena-text/src/main/java/org/apache/jena/query/text/TextIndexLucene.java:
```diff
@@ -704,7 +704,7 @@ private String composeQField(String qs, String textField, String lang, boolean u
                 textField += "_" + lang;
             }
-            textClause = textField + ":" + qs + " ";
+            textClause = textField + ": (" + qs + ") ";
```
Does the `usingSearchFor` branch of the `if` also need this fix?

OyvindLGjesdal replied:
Yes, I minimized the config, so it no longer has multilingual search active, which was the original use-case. Will add.

OyvindLGjesdal commented:
same bug applies also for usingSearchFor, see debug output for lucene on second test that is now added:

before fix

````
YIELDS: +(ftext_en-01:red comment:booze ftext_en-02:red comment:booze ftext_en-01:red comment:booze ftext_en-02:red comment:booze ftext_en-01:red comment:booze ftext_en-02:red comment:booze ftext_en-01:red comment:booze ftext_en-02:red comment:booze note_en-01:red comment:booze note_en-02:red comment:booze)

parsed query: +(ftext_en-01:red comment:booze ftext_en-02:red comment:booze ftext_en-01:red comment:booze ftext_en-02:red comment:booze ftext_en-01:red comment:booze ftext_en-02:red comment:booze ftext_en-01:red comment:booze ftext_en-02:red comment:booze note_en-01:red comment:booze note_en-02:red comment:booze) +graph:urn:x-arq:DefaultGraphNode; limit: 10000

```
after fix:

```
YIELDS: +((note_en-01:red note_en-01:booze) (note_en-02:red note_en-02:booze)) +graph:urn:x-arq:DefaultGraphNode; parsed query: +((ftext_en-01:red ftext_en-01:booze) (ftext_en-02:red ftext_en-02:booze) (ftext_en-01:red ftext_en-01:booze) (ftext_en-02:red ftext_en-02:booze) (ftext_en-01:red ftext_en-01:booze) (ftext_en-02:red ftext_en-02:booze) (ftext_en-01:red ftext_en-01:booze) (ftext_en-02:red ftext_en-02:booze) (note_en-01:red note_en-01:booze) (note_en-02:red note_en-02:booze)) +graph:urn:x-arq:DefaultGraphNode; limit: 10000

parsed query: +((ftext_en-01:red ftext_en-01:booze) (ftext_en-02:red ftext_en-02:booze) (ftext_en-01:red ftext_en-01:booze) (ftext_en-02:red ftext_en-02:booze) (ftext_en-01:red ftext_en-01:booze) (ftext_en-02:red ftext_en-02:booze) (ftext_en-01:red ftext_en-01:booze) (ftext_en-02:red ftext_en-02:booze) (note_en-01:red note_en-01:booze) (note_en-02:red note_en-02:booze))

```
The issue with possible duplicate expressions is there, but I don't think that changes the correctness?

OyvindLGjesdal commented:
I've attempted a fix for the duplicate expressions where the debug output looks good.

debug output:
*test01TextPropNotWorkingInSomeCases*, query 1:
before:
```
parsed query: +((ftext:red ftext:booze) (ftext:red ftext:booze) (ftext:red ftext:booze)
(ftext:red ftext:booze) (note:red note:booze)) +graph:urn:x-arq:DefaultGraphNode; 
limit: 10000
```
after:
```
parsed query: +((ftext:red ftext:booze) (note:red note:booze)) +graph:urn:x-arq:DefaultGraphNode; limit: 10000
```

*test02TextPropNotWorkingInSomeCases* query 1
before: 

```
parsed query: +((ftext_en-01:red ftext_en-01:booze) (ftext_en-02:red ftext_en-02:booze) (ftext_en-01:red ftext_en-01:booze) 
(ftext_en-02:red ftext_en-02:booze) (ftext_en-01:red ftext_en-01:booze) (ftext_en-02:red ftext_en-02:booze) 
(ftext_en-01:red ftext_en-01:booze) (ftext_en-02:red ftext_en-02:booze) (note_en-01:red note_en-01:booze)
(note_en-02:red note_en-02:booze)) +graph:urn:x-arq:DefaultGraphNode; limit: 10000
```
after:
```
parsed query: +((ftext_en-01:red ftext_en-01:booze) (ftext_en-02:red ftext_en-02:booze) (note_en-01:red note_en-01:booze) 
(note_en-02:red note_en-02:booze)) +graph:urn:x-arq:DefaultGraphNode; limit: 10000
```

OyvindLGjesdal commented:
When trying this branch in the manual actions job the PR breaks something (https://github.com/OyvindLGjesdal/jena/actions/runs/7324927309/job/19948943751)

it returns 114 errors.

```
23:21:45 WARN  TextIndexLuceneAssembler :: Multilingual support implicitly enabled by text:defineAnalyzers
Error:  Tests run: 303, Failures: 0, Errors: 114, Skipped: 8, Time elapsed: 4.547 s <<< FAILURE! - in org.apache.jena.query.text.TS_Text
Error:  org.apache.jena.query.text.TestDatasetWithLuceneTextIndex.propertyFunctionText_8  Time elapsed: 0.023 s  <<< ERROR!
org.apache.jena.query.text.TextIndexParseException: 
Text search parse error:
Cannot parse '': Encountered "<EOF>" at line 1, column 0.
Was expecting one of:
    <NOT> ...
    "+" ...
    "-" ...
    <BAREOPER> ...
    "(" ...
    "*" ...
    <QUOTED> ...
    <TERM> ...
    <PREFIXTERM> ...
    <WILDTERM> ...
    <REGEXPTERM> ...
    "[" ...
    "{" ...
    <NUMBER> ...
    <TERM> ...
    
	at org.apache.jena.query.text.TextIndexLucene.query(TextIndexLucene.java:503)
	at org.apache.jena.query.text.TextIndexLucene.query(TextIndexLucene.java:494)
	at org.apache.jena.query.text.TextQueryPF.performQuery(TextQueryPF.java:350)
	at org.apache.jena.query.text.TextQueryPF.lambda$query$1(TextQueryPF.java:303)
	at com.github.benmanes.caffeine.cache.BoundedLocalCache.lambda$doComputeIfAbsent$14(BoundedLocalCache.java:2688)
	at java.base/java.util.concurrent.ConcurrentHashMap.compute(ConcurrentHashMap.java:1916)
	at com.github.benmanes.caffeine.cache.BoundedLocalCache.doComputeIfAbsent(BoundedLocalCache.java:2686)
	at com.github.benmanes.caffeine.cache.BoundedLocalCache.computeIfAbsent(BoundedLocalCache.java:2669)
	at com.github.benmanes.caffeine.cache.LocalCache.computeIfAbsent(LocalCache.java:112)
	at com.github.benmanes.caffeine.cache.LocalManualCache.get(LocalManualCache.java:62)
	at org.apache.jena.atlas.lib.cache.CacheCaffeine.get(CacheCaffeine.java:89)
	at org.apache.jena.query.text.TextQueryPF.query(TextQueryPF.java:303)
	at org.apache.jena.query.text.TextQueryPF.prepareQuery(TextQueryPF.java:271)
	at org.apache.jena.query.text.TextQueryPF.exec(TextQueryPF.java:226)
	at org.apache.jena.sparql.pfunction.PropertyFunctionBase$RepeatApplyIteratorPF.nextStage(PropertyFunctionBase.java:104)
	at org.apache.jena.sparql.engine.iterator.QueryIterRepeatApply.makeNextStage(QueryIterRepeatApply.java:100)
	at org.apache.jena.sparql.engine.iterator.QueryIterRepeatApply.hasNextBinding(QueryIterRepeatApply.java:60)
	at org.apache.jena.sparql.engine.iterator.QueryIteratorBase.hasNext(QueryIteratorBase.java:116)
	at org.apache.jena.sparql.engine.iterator.QueryIterProcedure.hasNextBinding(QueryIterProcedure.java:73)
	at org.apache.jena.sparql.engine.iterator.QueryIteratorBase.hasNext(QueryIteratorBase.java:116)
	at org.apache.jena.sparql.engine.main.StageGeneratorGeneric.execute(StageGeneratorGeneric.java:56)
	at org.apache.jena.sparql.engine.main.StageGeneratorGeneric.execute(StageGeneratorGeneric.java:49)
	at org.apache.jena.tdb1.solver.StageGeneratorDirectTDB.execute(StageGeneratorDirectTDB.java:53)
	at org.apache.jena.tdb2.solver.StageGeneratorDirectTDB.execute(StageGeneratorDirectTDB.java:54)
	at org.apache.jena.sparql.engine.main.OpExecutor.execute(OpExecutor.java:126)
	at org.apache.jena.sparql.engine.main.ExecutionDispatch.visit(ExecutionDispatch.java:54)
	at org.apache.jena.sparql.algebra.op.OpBGP.visit(OpBGP.java:49)
	at org.apache.jena.sparql.engine.main.ExecutionDispatch.exec(ExecutionDispatch.java:43)
	at org.apache.jena.sparql.engine.main.OpExecutor.exec(OpExecutor.java:115)
	at org.apache.jena.sparql.engine.main.OpExecutor.execute(OpExecutor.java:225)
	at org.apache.jena.sparql.engine.main.ExecutionDispatch.visit(ExecutionDispatch.java:117)
	at org.apache.jena.sparql.algebra.op.OpSequence.visit(OpSequence.java:74)
	at org.apache.jena.sparql.engine.main.ExecutionDispatch.exec(ExecutionDispatch.java:43)
	at org.apache.jena.sparql.engine.main.OpExecutor.exec(OpExecutor.java:115)
	at org.apache.jena.sparql.engine.main.OpExecutor.execute(OpExecutor.java:393)
	at org.apache.jena.sparql.engine.main.ExecutionDispatch.visit(ExecutionDispatch.java:243)
	at org.apache.jena.sparql.algebra.op.OpProject.visit(OpProject.java:46)
	at org.apache.jena.sparql.engine.main.ExecutionDispatch.exec(ExecutionDispatch.java:43)
	at org.apache.jena.sparql.engine.main.OpExecutor.exec(OpExecutor.java:115)
	at org.apache.jena.sparql.engine.main.OpExecutor.execute(OpExecutor.java:87)
	at org.apache.jena.sparql.engine.main.QC.execute(QC.java:53)
	at org.apache.jena.sparql.engine.main.QueryEngineMain.eval(QueryEngineMain.java:55)
	at org.apache.jena.sparql.engine.QueryEngineBase.evaluate(QueryEngineBase.java:164)
	at org.apache.jena.sparql.engine.QueryEngineBase.createPlan(QueryEngineBase.java:128)
	at org.apache.jena.sparql.engine.QueryEngineBase.getPlan(QueryEngineBase.java:110)
	at org.apache.jena.sparql.engine.main.QueryEngineMain$QueryEngineMainFactory.create(QueryEngineMain.java:90)
	at org.apache.jena.sparql.engine.QueryEngineFactoryWrapper.create(QueryEngineFactoryWrapper.java:49)
	at org.apache.jena.sparql.exec.QueryExecDataset.getPlan(QueryExecDataset.java:514)
	at org.apache.jena.sparql.exec.QueryExecDataset.startQueryIterator(QueryExecDataset.java:455)
	at org.apache.jena.sparql.exec.QueryExecDataset.execute(QueryExecDataset.java:170)
	at org.apache.jena.sparql.exec.QueryExecDataset.select(QueryExecDataset.java:164)
	at org.apache.jena.sparql.exec.QueryExecutionAdapter.execSelect(QueryExecutionAdapter.java:106)
	at org.apache.jena.sparql.exec.QueryExecutionCompat.execSelect(QueryExecutionCompat.java:90)
	at org.apache.jena.query.text.AbstractTestDatasetWithTextIndexBase.doTestQuery(AbstractTestDatasetWithTextIndexBase.java:100)
	at org.apache.jena.query.text.AbstractTestDatasetWithTextIndexBase.doTestSearch(AbstractTestDatasetWithTextIndexBase.java:71)
	at org.apache.jena.query.text.AbstractTestDatasetWithTextIndexBase.doTestSearch(AbstractTestDatasetWithTextIndexBase.java:66)
	at org.apache.jena.query.text.AbstractTestDatasetWithTextIndexBase.doTestSearch(AbstractTestDatasetWithTextIndexBase.java:62)
	at org.apache.jena.query.text.AbstractTestDatasetWithTextIndex.propertyFunctionText_8(AbstractTestDatasetWithTextIndex.java:294)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:316)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:240)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:214)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:155)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)

Error:  org.apache.jena.query.text.TestDatasetWithLuceneTextIndex.propertyFunctionText_1_score  Time elapsed: 0.017 s  <<< ERROR!
org.apache.jena.query.text.TextIndexParseException: 
Text search parse error:
Cannot parse '': Encountered "<EOF>" at line 1, column 0.
Was expecting one of:
    <NOT> ...
    "+" ...
    "-" ...
    <BAREOPER> ...
    "(" ...
    "*" ...
    <QUOTED> ...
    <TERM> ...
    <PREFIXTERM> ...
    <WILDTERM> ...
    <REGEXPTERM> ...
    "[" ...
    "{" ...
    <NUMBER> ...
    <TERM> ...
    
	at org.apache.jena.query.text.TextIndexLucene.query(TextIndexLucene.java:503)
	at org.apache.jena.query.text.TextIndexLucene.query(TextIndexLucene.java:494)
	at org.apache.jena.query.text.TextQueryPF.performQuery(TextQueryPF.java:350)
	at org.apache.jena.query.text.TextQueryPF.lambda$query$1(TextQueryPF.java:303)
	at com.github.benmanes.caffeine.cache.BoundedLocalCache.lambda$doComputeIfAbsent$14(BoundedLocalCache.java:2688)
	at java.base/java.util.concurrent.ConcurrentHashMap.compute(ConcurrentHashMap.java:1916)
	at com.github.benmanes.caffeine.cache.BoundedLocalCache.doComputeIfAbsent(BoundedLocalCache.java:2686)
	at com.github.benmanes.caffeine.cache.BoundedLocalCache.computeIfAbsent(BoundedLocalCache.java:2669)
	at com.github.benmanes.caffeine.cache.LocalCache.computeIfAbsent(LocalCache.java:112)
	at com.github.benmanes.caffeine.cache.LocalManualCache.get(LocalManualCache.java:62)
	at org.apache.jena.atlas.lib.cache.CacheCaffeine.get(CacheCaffeine.java:89)
	at org.apache.jena.query.text.TextQueryPF.query(TextQueryPF.java:303)
	at org.apache.jena.query.text.TextQueryPF.prepareQuery(TextQueryPF.java:271)
	at org.apache.jena.query.text.TextQueryPF.exec(TextQueryPF.java:226)
	at org.apache.jena.sparql.pfunction.PropertyFunctionBase$RepeatApplyIteratorPF.nextStage(PropertyFunctionBase.java:104)
	at org.apache.jena.sparql.engine.iterator.QueryIterRepeatApply.makeNextStage(QueryIterRepeatApply.java:100)
	at org.apache.jena.sparql.engine.iterator.QueryIterRepeatApply.hasNextBinding(QueryIterRepeatApply.java:60)
	at org.apache.jena.sparql.engine.iterator.QueryIteratorBase.hasNext(QueryIteratorBase.java:116)
	at org.apache.jena.sparql.engine.iterator.QueryIterProcedure.hasNextBinding(QueryIterProcedure.java:73)
	at org.apache.jena.sparql.engine.iterator.QueryIteratorBase.hasNext(QueryIteratorBase.java:116)
	at org.apache.jena.sparql.engine.iterator.QueryIterConvert.hasNextBinding(QueryIterConvert.java:58)
	at org.apache.jena.sparql.engine.iterator.QueryIteratorBase.hasNext(QueryIteratorBase.java:116)
	at org.apache.jena.sparql.engine.iterator.QueryIteratorWrapper.hasNextBinding(QueryIteratorWrapper.java:38)
	at org.apache.jena.sparql.engine.iterator.QueryIteratorBase.hasNext(QueryIteratorBase.java:116)
	at org.apache.jena.sparql.engine.iterator.QueryIteratorWrapper.hasNextBinding(QueryIteratorWrapper.java:38)
	at org.apache.jena.sparql.engine.iterator.QueryIteratorBase.hasNext(QueryIteratorBase.java:116)
	at org.apache.jena.sparql.exec.RowSetStream.hasNext(RowSetStream.java:47)
	at org.apache.jena.sparql.engine.ResultSetStream.hasNext(ResultSetStream.java:75)
	at org.apache.jena.query.text.AbstractTestDatasetWithTextIndexBase.doTestQueryWithScores(AbstractTestDatasetWithTextIndexBase.java:125)
	at org.apache.jena.query.text.AbstractTestDatasetWithTextIndexBase.doTestSearchWithScores(AbstractTestDatasetWithTextIndexBase.java:76)
	at org.apache.jena.query.text.AbstractTestDatasetWithTextIndex.propertyFunctionText_1_score(AbstractTestDatasetWithTextIndex.java:113)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:316)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:240)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:214)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:155)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)

```

OyvindLGjesdal commented:
The PR is ready for review again, sorry for the noise.

the normal condition where propList isn't used failed, due to the *Remove duplicate Query expressions* commit overwriting qString, instead of appending.

```diff
+        qString += String.join("", uniquePropListQueryStrings);
-         qString = String.join("", uniquePropListQueryStrings);
```

Updated some variable names, forced updated to reasonable number of commits and CI job passed.

https://github.com/OyvindLGjesdal/jena/actions/runs/7366703795/job/20049203950

Also updated the PR description

rvesse commented on jena-text/src/main/java/org/apache/jena/query/text/TextIndexLucene.java:
```diff
@@ -722,6 +723,10 @@ private String composeQField(String qs, String textField, String lang, boolean u
 
+        // The set will reduce lucene text field expressions
+        // to unique expressions
+        Set<String> uniquePropListQueryStrings = new HashSet<>();
```
Is there any scenario in which the ordering of the generated clauses is going to matter?

i.e. should this be a `LinkedHashSet` instead?

+1 on eliminating duplicate clauses

OyvindLGjesdal replied:
Changed to `LinkedHashSet`
